### DISCUSSION TITLE
来个投票，期望 api 表中的弃用属性也走弃用过渡期

### DISCUSSION THREAD

**Discussion by Wxh16144** [ant-design/ant-design#51298] [Polls]
如题：期望 api 表中的弃用属性也走弃用过渡期（比如 3 - 5 个 minor 后才删除

比如我们加入了某个 feature，有些prop 应该弃用，现在是直接将弃用的某个属性从文档删除，仅仅在代码（控制台）中标记为弃用，感觉稍微有点不够。

提议如下改进：

> ~~我在这个 pr 实现了一版: https://github.com/ant-design/ant-design/pull/51297/files~~, 实现 PR： https://github.com/ant-design/ant-design/pull/51342

1. 在 markdown 中用 `~~{prop}~~` 标记弃用属性, 并添加一些样式。

```md
| ~~filterDropdownOpen~~ | 用于控制自定义筛选菜单是否可见, 请使用 `filterDropdownProps.open` 代替 | boolean | - | 4.23.0 |
```

<img width="1281" alt="image" src="https://github.com/user-attachments/assets/84b6d304-5910-45bf-8727-1eae56474fa5">

这样的话，需要在开发者在使用时才能发现问题。文档中的属性应该保留一段时间，以便开发者对应版本修改代码。

**Comment by Wxh16144** [comment]
直接删除的话，开发者拿着已弃用的属性在文档中搜索会搜索不到...

**Reply by afc163** [reply]
搜索不到是对的，不然太多冗余属性了。

**Reply by Wxh16144** [reply]
参考一下 mui 的，https://mui.com/material-ui/api/modal/
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/3453efb6-5e6c-4853-a806-2fd34d65e074">

（不过没有去深入了解他们的文档生成方式，貌似不是维护 markdown，而是 jsdoc 用脚本生成的元数据渲染的

**Reply by Wxh16144** [reply]
```css
tr:has(td:first-child > del) {
  color: ${token.colorWarning} !important;
  background-color: ${token.colorWarningBg} !important;
  td:first-child > del {
    color: ${token.colorWarning};
    border-bottom: 1px dashed ${token.colorWarningActive};
  }
}
```

**Comment by Zyf665** [comment]
这样确实感觉更友好些

**Comment by zombieJ** [comment]
看着效果不错，不过文档里删掉其实有意为之。如果要文档展示最好是有一个开关，显示废弃 API 之类的。

**Comment by yoyo837** [comment]
想法挺好，但，维护精力增加不少，好处不太明显。个人看法。

**Reply by Wxh16144** [reply]
以前已经删除的不用再加回去了，之后的操作只是 prop 加一个 `~~{}~~` ，感觉也还行。 然后某个 major 再去统一删除之类的

**Comment by afc163** [comment]
这个样式挺好的，1)放最后吧，2)默认不展示，加个已废弃 API 开关。

**Reply by Wxh16144** [reply]
预览链接：https://preview-51342-ant-design.surge.sh/components/input

<img width="1458" alt="image" src="https://github.com/user-attachments/assets/56bd0f4a-1f1a-45b8-bc6a-0f645e331253">
